designs/containers/workload-discovery/workload-discovery.md

```diff
@@ -49,10 +45,61 @@ The discovery flow adopted for this work will follow the same model already adop
 
 
 ### REST API  
-**TBD  on next revision**
 
-Describe in details any changes to the OSC REST APIs. This should include any new, modified or removed API and describing their payloads, headers and response status.
-> Note: Using the [swagger specification](#swagger-specification) is highly recommended.
+#### Security Groups
+The security group resource `/api/server/v1/virtualizationConnectors/{vcId}/securityGroups/{sgId}` is being slightly modified with:
+1.  **projectid** and **projectName** now are optional:  These fields are not applicable to Kubernetes.  For OpenStack, the enforcement of these fields will be done in the OSC business logic instead.
+2.  optional collection of **labels** being added: These fields are applicable only to Kubernetes and the following restrictions will be enforced by the OSC busines logic for Kubernetes security groups:   
+a) At least one label must be provided when creating or updating a security group;  
+b) All labels must be strings in the format "key=value". 
+
+> Note: The choice of using a string rather a well defined data type for labels with {key; value} is because the key+value format may not be applicable to other virtualization platforms we may support in the future. 

```

Comment:
I emailed Tarun and Manish to have this simplified further for a single label. If that is ok then this will likely just be a string field. Selecting pods in k8s by a single label does not require the knowledge of the format "key=value", this is only needed if one is implementing and OR selector. I will update this once I get an answer.